### PR TITLE
[git-release-manager plugin] Introduce ability to add custom features

### PR DESCRIPTION
**Pull request by erikengervall**
Add more metadata to success callbacks

## Hey, I just made a Pull Request!

Introducing the ability for users to add their own features to git-release-manager. This is very handy if you have needs beyond the standard features and would like to leverage that plenty of information is readily available (e.g. repository information, project information and so on).

## Note

When introducing this change I relied heavily on the checklist below. Following the link in the first item (`A changeset describing the change and affected packages.`) gave me useful information about changesets, but it doesn't mention whether or not one has to bump ones `package.json` manually nor if the newly bumped version will be published to npm 🤔 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by regicsolutions** [comment]
Cool! Any plans to enable support for other SCM providers such as Bitbucket?

File: plugins/git-release-manager/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@backstage/plugin-git-release-manager",
-  "version": "0.1.3",
+  "version": "0.2.0",
```
Comment (OrkoHunter) [inline plugins/git-release-manager/package.json:3]: Hmm. This version bump is something that changeset bot handles, along with generating the changelog and publishing the package to npm. It's done when this Version Packages PR is merged. https://github.com/backstage/backstage/pull/6507

So, I don't think you need to change the version, as long as you have a changeset in this PR. :)

Comment (erikengervall) [reply plugins/git-release-manager/package.json:3]: I had a suspicious, but wasn't sure 😅

I'll revert this bump :)

Comment (OrkoHunter) [reply plugins/git-release-manager/package.json:3]: Noice!

**Comment by erikengervall** [comment]
> Cool! Any plans to enable support for other SCM providers such as Bitbucket?

I can't speak for Backstage's support (I do know there's plenty of [integrations in the docs](https://backstage.io/docs/integrations/), but haven't experimented with them myself), but from the Plugin's point of view it is SCM agnostic in the sense that one can supply their own ApiClient by implementing the API's [interface](https://github.com/backstage/backstage/blob/91088c38774f4da3328196451bf9c3d0a9e9962a/plugins/git-release-manager/src/api/GitReleaseClient.ts#L549).

**Review by OrkoHunter — APPROVED** [review]
LGTM!

**Comment by erikengervall** [comment]
> LGTM!

Could you help me merge it, please? 🙂

**Comment by regicsolutions** [comment]
I already have this set in my api.ts in the front end tried and tried installing it with Bitbucket Server and am getting prompted to authenticate using GitHub auth, is there a way to override the auth?

**Comment by erikengervall** [comment]
> I already have this set in my api.ts in the front-end and tried installing it with Bitbucket Server and am getting prompted to authenticate using GitHub auth, is there a way to override the auth?

tbh haven’t had time to experiment with other providers yet, but I’ve poked the backstage folks about this and will continue working with the plugin after vacay 🙂

**Pull request merged**